docker/test/integration/cluster/ImageStore.py

```diff
@@ -97,10 +105,21 @@ def __build_minifi_cpp_sql_image(self):
 
         return self.__build_image(dockerfile)
 
-    def __build_minifi_cpp_image_with_nifi_python_processors(self, additional_cmd=""):
+    def __build_minifi_cpp_image_with_nifi_python_processors(self, python_option):
         parse_document_url = "https://raw.githubusercontent.com/apache/nifi/rel/nifi-" + NifiContainer.NIFI_VERSION + "/nifi-python-extensions/nifi-text-embeddings-module/src/main/python/ParseDocument.py"
         chunk_document_url = "https://raw.githubusercontent.com/apache/nifi/rel/nifi-" + NifiContainer.NIFI_VERSION + "/nifi-python-extensions/nifi-text-embeddings-module/src/main/python/ChunkDocument.py"
         pip3_install_command = ""
+        requirements_install_command = ""
+        additional_cmd = ""
+        parse_document_sed_cmd = 'sed -i "54d;55d" /opt/minifi/minifi-current/minifi-python/nifi_python_processors/ParseDocument.py && \\'
+        chunk_document_sed_cmd = 'sed -i "112d" /opt/minifi/minifi-current/minifi-python/nifi_python_processors/ChunkDocument.py && \\'
+        if python_option == PythonOptions.SYSTEM_INSTALLED_PACKAGES:
+            additional_cmd = "RUN pip3 install 'langchain<=0.17.0'"
+        elif python_option == PythonOptions.REQUIREMENTS_FILE:
+            requirements_install_command = "echo 'langchain<=0.17.0' > /opt/minifi/minifi-current/minifi-python/nifi_python_processors/requirements.txt && \\"
+        elif python_option == PythonOptions.INLINE_DEFINED_PACKAGES:
+            parse_document_sed_cmd = parse_document_sed_cmd[:-2] + ' sed -i "54 i \\ \\ \\ \\ \\ \\ \\ \\ dependencies = [\\"langchain<=0.17.0\\"]" /opt/minifi/minifi-current/minifi-python/nifi_python_processors/ParseDocument.py && \\'
```

Comment:
Shouldn't there be 3 backslashes around the langchain dependency? 2 for a single escaped backslash for sed, and a third for escaping the quotation mark, so it goes to sed, not the shell.

Reply:
I think you are right, I tested and it also works so I updated it in 327ddca15c6bf2f9692d3da061f5688596537c06